### PR TITLE
Feature: Recipe List

### LINKED ISSUE #3: [3pt] Recipes List
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.

### PR DESCRIPTION
**Pull request by leonardodiasb**
## This PR adds:

- Display of recipes(Recipes[index])
- Add new recipes(Recipes[new create])
- Delete recipes(Recipes[destroy])

Additional: 
- Model validations for recipe fields.
- Added `configure_permitted_parameters` method to `ApplicationController`.
- Added `authenticate_user` method to `FoodsController`, `RecipeFoodsController`, `RecipesController`, `ShoppingController` and `UsersController`.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
   def index; end
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name posts_counter])
```
Comment (D4nielJ) [inline app/controllers/application_controller.rb]: I think you missed the `posts_counter` variable from the last project here 😅

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -1,3 +1,41 @@
 class RecipesController < ApplicationController
-  def index; end
+  before_action :authenticate_user!
+
+  def index
+    @recipes = Recipe.all
```
Comment (D4nielJ) [inline app/controllers/recipes_controller.rb]: I think we should retrieve only the recipes of the current_user in the index of this controller. That way we won't be able to see recipes from other users in our own list of recipes.

We will have a public recipes list later on.

**Review by D4nielJ — COMMENTED** [review]
## Nice job!! 🔥 🔥 

What do you think of my comments? Let me know.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (leonardodiasb) [reply app/controllers/application_controller.rb]: Sorry for that. I will fix it! 🤦‍♂️

File: app/controllers/recipes_controller.rb (hunk 2 above)
Comment (leonardodiasb) [reply app/controllers/recipes_controller.rb]: You're right! 👍

**Review by D4nielJ — APPROVED** [review]

**Pull request merged**